pull.go

```diff
@@ -112,9 +112,14 @@ func (c *Client) fetch(ctx context.Context, rCtx *RemoteContext, ref string, lim
 		childrenHandler := images.ChildrenHandler(store)
 		// Set any children labels for that content
 		childrenHandler = images.SetChildrenLabels(store, childrenHandler)
-		// Filter manifests by platforms but allow to handle manifest
-		// and configuration for not-target platforms
-		childrenHandler = remotes.FilterManifestByPlatformHandler(childrenHandler, rCtx.PlatformMatcher)
+		if rCtx.AppendDistributionSourceLabel {
```

Comment:
I think this is worthy of a new option for all manifests or all image metadata. We haven't finalized cross repository push so it is possible we want to append the distribution labels without pulling each manifest.